### DISCUSSION TITLE
config/settings is missing

### DISCUSSION THREAD

**Discussion by drmad** [nginx/unit#1159] [Support] [answered]
Hi.

I build unit 1.31.1-1 from source. I can work fine manipulating `config/listeners`, `config/applications`, `config/routes`, etc. but I can't access `config/settings`. A `GET /config/settings` results in a 404 `{"error": "Value doesn't exist."}`, and trying to POST something inside also fails with 404.

Am I missing something?

**Comment by tippexs** [comment, marked as answer]
Hi @drmad Thanks for reaching out!

This is how you can initialize the `settings` object
```bash
$  curl --unix-socket /var/run/control.unit.sock -X PUT -d '{}' localhost/config/settings/
{
	"success": "Reconfiguration done."
}
```
Sorry for this bad experience! We will think about this and will come up with an idea on how to make this better in the future!

**Reply by drmad** [reply]
Hi. Thanks for the info. 

I suppose I have to create all the structures in `settings` in order to reach, for instance, `config/settings/http/log_route`. I'll try 👍

**Comment by ac000** [comment]
I guess that's because we only initialise the listeners, routes and applications objects at startup.

From `src/nxt_controller.c::nxt_controller_conf_default()`

```c
513    static const nxt_str_t json = nxt_string(
514        "{ \"listeners\": {}, \"routes\": [], \"applications\": {} }"
515    );
```

**Comment by lcrilly** [comment]
It's a combination of the default configuration is (per @ac000 's comment) and the fact that

Unit's control API does not create parent objects automatically.

At the very least, this behaviour should be documented at https://unit.nginx.org/controlapi/
